src/lib/services/segment-service.ts

```diff
@@ -82,8 +84,13 @@ export class SegmentService {
     async update(id: number, data: unknown, user: User): Promise<void> {
         const input = await segmentSchema.validateAsync(data);
         this.validateSegmentValuesLimit(input);
-
         const preData = await this.segmentStore.get(id);
+
+        if (preData.name !== input.name) {
+            // If the name has changed, make sure it's still unique
```

Comment:
Given the context I find the code here pretty clear. I usually recommend comments when domain logic is not obvious or there's a hidden trap where the obvious thing is subtly incorrect. Totally fine with you leaving this in, but going forward let's keep to the code base style here, which is use comments sparingly currently